pkg/databind/pkg/databind/binder_test.go

```diff
@@ -98,3 +99,239 @@ func TestContextCache(t *testing.T) {
 	result = fetch()
 	assert.Equal(t, fetched{"bye", "bye", "bye"}, result)
 }
+
+func mockGatherer(ttl time.Duration, data interface{}) *gatherer {
+	return &gatherer{
+		cache: cachedEntry{ttl: ttl}, //nolint:exhaustruct
+		fetch: func() (interface{}, error) {
+			return data, nil
+		},
+	}
+}
+
+// dataWithTTL is a custom implementation of a payload exposing a TTL.
+type dataWithTTL map[string]interface{}
+
+func (ttl dataWithTTL) TTL() (time.Duration, error) {
+	var ok bool //nolint:varnamelen
+	var ttlData string
+
+	if _, ok = ttl["ttl"]; !ok {
+		return 0, ErrTTLNotFound
+	}
+
+	if ttlData, ok = ttl["ttl"].(string); !ok {
+		return 0, ErrTTLInvalid
+	}
+
+	t, err := time.ParseDuration(ttlData)
+	if err != nil {
+		return 0, multierr.Append(ErrTTLInvalid, err)
+	}
+
+	return t, nil
+}
+
+func (ttl dataWithTTL) Data() (map[string]interface{}, error) {
+	if _, ok := ttl["data"]; !ok {
+		return nil, ErrDataNotFound
+	}
+
+	if _, ok := ttl["data"].(map[string]interface{}); !ok {
+		return nil, ErrDataInvalid
+	}
+
+	//nolint:forcetypeassert
+	return ttl["data"].(map[string]interface{}), nil
+}
+
+func Test_GathererCacheTtlFromPayload(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name               string
+		cacheInitialTTL    time.Duration
+		mockData           interface{}
+		expectedTTLInCache time.Duration
+		expectedError      error
+	}{
+		{
+			name:            "no ttl implementation should respect original ttl",
```

Comment:
Why is ttl not implemented in this case? Because no data was provided?

Reply:
This case is using a `map[string]interface{}` that does not implement the interface but it has a field with name `ttl` . It's just to assure that the field is not taking into accountl.